src/main/java/org/openrewrite/java/testing/jmockit/JMockitExpectationsToMockito.java

```diff
@@ -166,29 +173,80 @@ private J.Block applyTemplate(ExecutionContext ctx, List<Object> templateParams,
                     );
         }
 
-        private void rewriteArgumentMatchers(ExecutionContext ctx, List<Object> templateParams) {
-            J.MethodInvocation invocation = (J.MethodInvocation) templateParams.get(0);
+        private void rewriteArgumentMatchers(ExecutionContext ctx, List<Object> bodyTemplateParams) {
+            J.MethodInvocation invocation = (J.MethodInvocation) bodyTemplateParams.get(0);
             List<Expression> newArguments = new ArrayList<>(invocation.getArguments().size());
             for (Expression methodArgument : invocation.getArguments()) {
                 if (!isArgumentMatcher(methodArgument)) {
                     newArguments.add(methodArgument);
                     continue;
                 }
-                String argumentMatcher = ((J.Identifier) methodArgument).getSimpleName();
-                maybeAddImport("org.mockito.Mockito", argumentMatcher);
-                newArguments.add(JavaTemplate.builder(argumentMatcher + "()")
-                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "mockito-core-3.12"))
-                        .staticImports("org.mockito.Mockito." + argumentMatcher)
-                        .build()
-                        .apply(
-                                new Cursor(getCursor(), methodArgument),
-                                methodArgument.getCoordinates().replace()
-                        ));
+                String argumentMatcher, template;
+                List<Object> argumentTemplateParams = new ArrayList<>();
+                if (!(methodArgument instanceof J.TypeCast)) {
+                    argumentMatcher = ((J.Identifier) methodArgument).getSimpleName();
+                    template = argumentMatcher + "()";
+                    newArguments.add(rewriteMethodArgument(ctx, argumentMatcher, template, methodArgument,
+                            methodArgument.getCoordinates().replace(), argumentTemplateParams));
+                    continue;
+                }
+                J.TypeCast tc = (J.TypeCast) methodArgument;
+                argumentMatcher = ((J.Identifier) tc.getExpression()).getSimpleName();
+                String className, fqn;
+                JavaType typeCastType = tc.getType();
+                if (typeCastType instanceof JavaType.Parameterized) {
+                    // strip the raw type from the parameterized type
+                    className = ((JavaType.Parameterized) typeCastType).getType().getClassName();
+                    fqn = ((JavaType.Parameterized) typeCastType).getType().getFullyQualifiedName();
+                } else if (typeCastType instanceof JavaType.FullyQualified) {
+                    className = ((JavaType.FullyQualified) typeCastType).getClassName();
+                    fqn = ((JavaType.FullyQualified) typeCastType).getFullyQualifiedName();
+                } else {
+                    throw new IllegalStateException("Unexpected value: " + typeCastType);
```

Comment:
We prefer not to throw exception, and instead not to make a change; would that be possible here?
That way recipes are safe to run as part of a larger set, only making changes when safe to do so.

Reply:
@timtebeek 

I added a try/catch around the entire recipe to that if anything goes wrong we simply return the original LST.